packages/nouns-contracts/contracts/governance/ExcessETHBurner.sol

```diff
@@ -0,0 +1,323 @@
+        uint256 nounId,
+        uint64 initialBurnNounId_,
+        uint64 nounIdsBetweenBurns_
+    ) public view returns (bool) {
+        if (nounId < initialBurnNounId_) return false;
+
+        uint256 distanceFromBurnWindowStart = (nounId - initialBurnNounId_) % nounIdsBetweenBurns_;
+
+        return distanceFromBurnWindowStart <= burnWindowSize;
+    }
+
+    /**
+     * @notice Returns the next burn window start id.
+     * For a given `currentNounId`, it returns the next id which can be represented by
+     * `initialBurnNounId_` + N * `nounIdsBetweenBurns_` for any N >= 0.
+     */
+    function nextBurnWindowStart(
+        uint64 currentNounId,
+        uint64 initialBurnNounId_,
+        uint64 nounIdsBetweenBurns_
+    ) public pure returns (uint64) {
+        // this could not happen during a burn. here as convenience.
+        if (currentNounId < initialBurnNounId_) return initialBurnNounId_;
+
+        uint64 distanceFromBurnWindowStart = (currentNounId - initialBurnNounId_) % nounIdsBetweenBurns_;
+        return currentNounId - distanceFromBurnWindowStart + nounIdsBetweenBurns_;
+    }
+
+    /**
+     * @notice Get the amount of excess ETH in the Nouns treasury.
+     * Excess ETH is defined as the difference between the current treasury value denominated in ETH,
+     * and the expected treasury value denominated in ETH, which is defined as mean auction price * adjusted total supply.
+     * It returns zero during the waiting period, and when expected treasury value is greater than current value.
+     * If treasury balance in native ETH is less than the excess ETH calculation, it returns the treasury ETH balance.
+     * @dev This version does not support burning excess other than in native ETH, e.g. stETH, rETH, etc.
+     * @dev Reverts if there is not enough auction history to calculate the mean auction price.
+     */
+    function excessETH() public view returns (uint256) {
+        uint256 expectedTreasuryValue = expectedTreasuryValueInETH();
+        uint256 treasuryValue = treasuryValueInETH();
+
+        if (expectedTreasuryValue >= treasuryValue) return 0;
+
+        return min(treasuryValue - expectedTreasuryValue, owner().balance);
+    }
+
+    /**
+     * @notice Get the expected treasury value denomiated in ETH.
+     * Expected value is defined as mean auction price * adjusted total supply.
+     */
+    function expectedTreasuryValueInETH() public view returns (uint256) {
+        return meanAuctionPrice() * dao.adjustedTotalSupply();
+    }
+
+    /**
+     * @notice Get the current treasury value denomiated in ETH.
+     * In addition to native ETH, it also includes stETH, wETH, and rETH balances.
+     * @dev for rETH, it uses the getEthValue() function to convert rETH to ETH.
+     */
+    function treasuryValueInETH() public view returns (uint256) {
+        address owner_ = owner();
+        return owner_.balance + stETH.balanceOf(owner_) + wETH.balanceOf(owner_) + rETHBalanceInETH(owner_);
```

Comment:
Thoughts on including token buyer's ETH in this calculation as well? 10% of the treasury currently sits in the token buyer.

Reply:
let's discuss this.
to consider: token buyer's ETH is not part of the forked assets, so it doesn't participate in the BV

Reply:
@solimander we haven't considered including token buyer's ETH in the calculation so far.
agree that currently a big chunk of the treasury is there, but I think that's an anomaly rather than the norm. We're got to much ETH sent there because many were looking at the state of token buyer as a candidate and those transactions haven't since updated. I feel like this is a more of a bug with token buyer rather than a normal state. Perhaps token buyer should send ETH back to the DAO if it has too much? haven't thought about it enough.

Regardless, I think there's some logic in excluding it from the calculation here, because it doesn't participate in the assets sent to a fork. I don't feel strongly about it, so curious to hear your thoughts here.
also @eladmallel

Reply:
Okay, makes sense. I agree it likely doesn't make sense to add since it's not included in BV, but we should probably address this issue at some point.

Reply:
I thought about it for a brief moment some weeks ago and decided to leave it out for the reason you mentioned - since that ETH is anyway currently not part of the forking mechanism.

but since it has the potential to hold significant amounts of ETH, and has the potential to be used to DAO spending, it can be a weird way of gaming the burn against our intentions, and because it seems fairly simple to take it into account - I am now leaning towards including its ETH in this calculation.

@davidbrai let's just add it?